cep-20.2.md

```diff
@@ -0,0 +1,829 @@
+  secrets: [string]
+  # The file to use as the script. Automatically adds `bat` or `sh` to the filename
+  # on Windows or UNIX respectively (if no file extension is given).
+  file: string  # build.sh or build.bat
+  # A string or list of strings that is the script contents (mutually exclusive with `file`)
+  content: string | [string]
+```
+
+## Source section
+
+```yaml
+source: [SourceElement]
+```
+
+where the different source elements are defined as follows.
+
+### URL source
+
+```yaml
+# url pointing to the source tar.gz|zip|tar.bz2|... (this can be a list of mirrors that point to the same file)
+url: url | [url]
+# destination folder in work directory
+target_directory: path
+# rename the downloaded file to this name
+file_name: string
+# hash of the file
+sha256: hex string
+# legacy md5 sum of the file (test both, prefer sha256)
+md5: hex string
+# relative path from recipe file
+patches: [path]
+```
+
+### Local source
+
+A path can be either:
+
+- directory ("../bla")
+- a path to an archive ("../bla.tar.gz")
+- a path to a file ("../bla.txt")
+
+```yaml
+# file, absolute or relative to recipe file
+path: path
+# if there is a gitignore, adhere to it (or not)
+use_gitignore: bool (defaults to true)
+# destination folder
+target_directory: path
+# rename the downloaded file to this name
+file_name: string
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Git source
+
+```yaml
+# URL to the git repository or path to local git repository
+git: url | path
+# the following 3 keys are mutually exclusive (branch, tag, and rev)
+# branch to checkout to
+branch: string
+# tag to checkout to
+tag: string
+# revision to checkout to (hash or ref)
+rev: string
+# depth of the git clone (mutually exclusive with rev)
+depth: signed integer (defaults to -1 -> not shallow)
+# should this use git-lfs?
+lfs: bool (defaults to false)
+# destination folder in work directory
+target_directory: path
+# absolute or relative path from recipe file
+patches: [path]
+```
+
+### Removed source definitions
+
+SVN and HG (mercury) source definitions are removed as they are not relevant anymore.
+
+## Requirements section
+
+```yaml
+requirements:
+  # build time dependencies, in the build_platform architecture
+  build: [PackageSelector]
+  # dependencies to link against, in the target_platform architecture
+  host: [PackageSelector]
+  # the section below is copied into the index.json and required at package installation
+  run: [PackageSelector]
+  # constrain optional packages (was `run_constrained`)
+  run_constraints: [PackageSelector]
+  # the run exports of this package
+  run_exports: [PackageSelector] OR RunExports
+
+  # the run exports to ignore when calculating the requirements
+  ignore_run_exports:
+    # ignore run exports by name (e.g. `libgcc-ng`)
+    by_name: [string]
+    # ignore run exports that come from the specified packages
+    from_package: [string]
```

Comment:
Why would you think it refers to a MatchSpec?
It refers to a single package _name_ (string). Same for both of these options.

Reply:
I fail to see the use case for a matchspec here.

Reply:
Oh, I'm not suggesting to use `MatchSpec`.
I just wanted to note that `conda-build` accepts more than just the name in those fields.
Looking at the code at
- https://github.com/conda/conda-build/blob/3.28.0/conda_build/render.py#L418
- https://github.com/conda/conda-build/blob/3.28.0/conda_build/render.py#L238
it seems to (effectively with some manual string handling) mostly concern itself with the names too.
We should just look out for such oddities so that any incompatibility between the old and new recipe formats are deliberate.

----
> Why would you think it refers to a MatchSpec?
> It refers to a single package _name_ (string). Same for both of these options.

I'd hope it refers to just the names :).
Since there is a naming inconsistency (`_name`/`_package`) is it not unreasonable to wonder if there is a behavioral difference too, is all I'm saying.